cmd/otelcorecol/README.md

```diff
@@ -0,0 +1,4 @@
+# `otelcorecol` test binary
+
+This folder contains the sources for the `otelcorecol` test binary. This binary is intended for **TEST PURPOSES ONLY**. The source files in this folder are **NOT** the ones used to build any official OpenTelemetry Collector releases.
```

Comment:
You could add that this is about integration tests in this repo and not for evaluation purposes by end users, e.g. like

```suggestion
This folder contains the sources for the `otelcorecol` test binary. This binary is intended for internal **TEST PURPOSES ONLY**. The source files in this folder are **NOT** the ones used to build any official OpenTelemetry Collector releases.
```